### PR TITLE
[webpack]: use path.resolve instead of require

### LINKED ISSUE #12837: TypeError: Cannot read properties of undefined in Content-Type Builder and Uploads pages
## Bug report
Hello,

I've an issue in the "Content-Type Build" page with the error `Cannot read properties of undefined (reading 'startSection')`
and the "uploads" page (under the content-type build) with the error `Cannot read properties of undefined (reading 'allPermissions')`

### Describe the bug
 When I'm going on the url `127.0.0.1/admin/plugins/content-type-builder` i get a blank page (here black because of my browser dark mode) with this error in the console
 
(c.f Screen 1)

Also the same problem but with another property in the uploads pages (`127.0.0.1/admin/plugins/upload`)

(c.f Screen 2)

I tried with version 4.0.5, 4.1.0, 4.1.1, 4.1.2 & 4.1.3
Each time I removed the package-lock.json, removed the `build` and `.cache` folders, and then start the server and I always have the same problem on the both pages.

### Steps to reproduce the behavior

1. Go to admin panel
2. Click on Content-Type Builder
4. See blank page & error
5. Go to admin panel index by refreshing or opening a new tab
6. Click on "Media library"
7. See blank page & error

### Expected behavior

Accessing both content type builder and media library page without errors in console and blank page.

### Screenshots

Screen 1 : image
Screen 2: image

### System

- Node.js version: v14.19.0
- NPM version: v14.19.0
- Strapi version: Tested on 4.0.5, 4.1.0, 4.1.1, 4.1.2 & 4.1.3
- Database: Mysql 8.0.28
- Operating system: Ubuntu 18.04.6 LTS (GNU/Linux 4.15.0-171-generic x86_64)

### Additional context

The problem appears after I got for an unknown reason a deletion of the content of the strapi folder, after reverting it with git, the problem was here.

### PR DESCRIPTION
**Pull request by joshuaellis**
### What does it do?

Replaces usage of `require.resolve` with `path.resolve` in the admin webpack alias-ing config part.

### Why is it needed?

Because `require.resolve` demands the `CJS` version of a library therefore not allowing the ESM to be used in bundling (which is typically better treeshaked)

### How to test it?

I'd like this to be released as a `beta` on wednesday so we can discuss with some members of the community to solve their issues

### Related issue(s)/PR(s)

* Is needed for #14821 to be merged
* Hopefully resolves #12837

### CODE REVIEW

File: packages/core/admin/webpack.alias.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ const aliasExactMatch = [
 // See https://webpack.js.org/configuration/resolve/
 module.exports = {
   ...aliasExactMatch.reduce((acc, name) => {
-    acc[`${name}$`] = require.resolve(name);
+    acc[`${name}$`] = path.resolve(__dirname, '..', '..', '..', 'node_modules', name);
```
Comment (alexandrebodin) [inline packages/core/admin/webpack.alias.js:43]: We probably want to use sth like findup to find the package up the tree if Strapi is in a monorepo senario

**Comment by alexandrebodin** [comment]
> @alexandrebodin I've opted to use `find-root` because we can pass it the string we were using before (the CJS module) and it will relatively find the package.json relative to that file therefore allowing the bundler to pick appropriately the correct file type.
> 
> More confident in this solution going forward as well because it's extremely close to our current implementation.

Interesting it looks like it's doing something like we do in the backend here packages/core/strapi/lib/load/package-path.js. Could we use that method ?

File: packages/core/admin/package.json [hunk 2]
```diff
@@ -72,7 +72,6 @@
     "esbuild-loader": "^2.20.0",
     "execa": "^1.0.0",
     "fast-deep-equal": "3.1.3",
-    "find-root": "1.1.0",
```
Comment (alexandrebodin) [inline packages/core/admin/package.json]: can remove it from here and yarn then

Comment (joshuaellis) [reply packages/core/admin/package.json]: That is it being removed :) I think another package in the repo needs it hence why it still shows in the `yarn.lock`.

**Comment by joshuaellis** [comment]
@Convly I believe this is ready for an experimental release. Do you want to leave this PR, add `dont merge` flag and release from this branch or do you want to merge it into a different branch? or something else? 😄

**Comment by Convly** [comment]
If you can update this branch with main and add a "do not merge" label, we can release an experimental from this branch :slightly_smiling_face:

**Comment by Convly** [comment]
@joshuaellis 

A new experimental release has been deployed for `chore/use-path-instead-of-require` under the `0.0.0-b4eaa97842f53ed5611c878ecef9654b7bde1c11` version.

To create a new application, use the following command
```sh
npx create-strapi-app@0.0.0-b4eaa97842f53ed5611c878ecef9654b7bde1c11 my-app --quickstart
```

**Comment by joshuaellis** [comment]
@gu-stav can you review this please? I think we can merge it now 👍🏼

**Review by gu-stav — APPROVED** [review]

**Review by alexandrebodin — APPROVED** [review]

**Pull request merged**